dgc-testdata-tests/DGCTestDataVerifier.cs

```diff
@@ -0,0 +1,81 @@
+﻿using DCC;
+using DGC;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+
+namespace dgc.testdata.tests
+{
+    public class DGCTestDataVerifier
+    {
+        record TestData(string prefix, string certificate, string filename);
+
+        public record EXPECTEDRESULTS(bool EXPECTEDVALIDOBJECT, bool EXPECTEDSCHEMAVALIDATION, bool EXPECTEDENCODE,
+            bool EXPECTEDDECODE, bool EXPECTEDVERIFY, bool EXPECTEDCOMPRESSION,
+            bool EXPECTEDKEYUSAGE, bool EXPECTEDUNPREFIX, bool EXPECTEDVALIDJSON,
+            bool EXPECTEDB45DECODE, bool EXPECTEDPICTUREDECODE, bool EXPECTEDEXPIRATIONCHECK);
+        public record TESTCTX(string VERSION, string SCHEMA, string CERTIFICATE, DateTime VALIDATIONCLOCK, string DESCRIPTION);
+        public record TestDataStructure(JObject JSON, string CBOR, string COSE, string COMPRESSED, string BASE45, string PREFIX, string x2DCODE, TESTCTX TESTCTX, EXPECTEDRESULTS EXPECTEDRESULTS);
+
+        public static async Task TestAll()
+        {
+            var jsonfiles = Directory.GetFiles(@"./dgc-testdata", "*.json", SearchOption.AllDirectories);
+
+            var testdataset = new List<Tuple<TestDataStructure, string>>();
+            foreach (var jsonTestDataFile in jsonfiles)
+            {
+                using var file = File.OpenText(jsonTestDataFile);
+                try
+                {
+                    var jsonStr = file.ReadToEnd();
+
+                    var test = JsonConvert.DeserializeObject<TestDataStructure>(jsonStr);
+
+                    testdataset.Add(Tuple.Create(test, jsonTestDataFile));
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine(file + ": " + ex.Message);
+                }
+            }
+
+            var dcgDecoder = new GreenCertificateDecoder();
+            var secretariat = new SecretariatService();
+            var verifier = new GreenCertificateVerifier(secretariat);
+            foreach (var testdataAndFile in testdataset)
+            {
+                var testdata = testdataAndFile.Item1;
+                var errorMessage = "Test Passes";
```

Comment:
minor: probably set errorMessage - if at all - to Test Fails and only set to Test Passes if all is known to be good?